Sources/EdgeConsentStatus.swift

```diff
@@ -12,7 +12,37 @@
 
 import Foundation
 
-enum ConsentStatus: String, Codable, Equatable {
+enum ConsentStatus: String, RawRepresentable, Codable {
     case yes = "y"
     case no = "n"
+    case pending = "p"
+
+    typealias RawValue = String
+
+    /// Initializes the appropriate `ConsentStatus` enum for the given `rawValue`
+    /// - Parameter rawValue: a `RawValue` representation of a `ConsentStatus` enum, default is pending
+    public init(rawValue: RawValue) {
+        switch rawValue {
+        case "y":
+            self = .yes
+        case "n":
+            self = .no
+        default:
```

Comment:
The default case is a bit confusing for me, as we have two default values in our constants based on specific scenarios. I wonder if it's the correct thing to do to default to pending here.

What if we made it a failable initializer and returned nil instead of providing a default value as it can vary based on the situation?

`init?(rawValue: Self.RawValue)`

Reply:
Don't think it will help much because I am already unwrapping the string before calling CollectConsent(rawValue:) - see getCollectConsentOrDefault. Changed to return the Defaults.COLLECT_CONSENT_PENDING.

The only case where we use default yes is on bootup when Consent extension is not registered.